src/librustc_mir/transform/add_yk_swt_calls.rs

```diff
@@ -175,9 +175,8 @@ fn is_untraceable(tcx: TyCtxt<'a, 'tcx, 'tcx>, src: MirSource) -> bool {
         return true;
     }
 
-    // FIXME Anything which the trace recorder depends upon cannot be traced or we will get
-    // infinite recursion at runtime..
-    if RECORDER_DEPS.contains(&&*tcx.crate_name(LOCAL_CRATE).as_str()) {
+    // Don't transform any crate which is blacklisted. See the comment against BLACKLISTED_CRATES.
```

Comment:
I don't think we need this comment (it just says what the line of code below does, and the line of code is very clear IMHO).